src/tools/run-make-support/src/fs.rs

```diff
@@ -50,7 +14,34 @@ pub fn copy_dir_all(src: impl AsRef<Path>, dst: impl AsRef<Path>) {
             if ty.is_dir() {
                 copy_dir_all_inner(entry.path(), dst.join(entry.file_name()))?;
             } else if ty.is_symlink() {
-                copy_symlink(entry.path(), dst.join(entry.file_name()))?;
+                // Traverse symlink once to find path of target entity.
+                let target_path = std::fs::read_link(entry.path())?;
+
+                let new_symlink_path = dst.join(entry.file_name());
+                #[cfg(windows)]
+                {
+                    // Find metadata about target entity (shallow, no further symlink traversal in
+                    // case target is also a symlink).
+                    let target_metadata = std::fs::symlink_metadata(&target_path)?;
```

Comment:
We could just use `ty.is_symlink_dir()` or `ty.is_symlink_file()`. We're copying here so it's not like there be a difference between the two (or if there is then something is already badly broken).

Reply:
Changed, thanks.